# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1436836908 14400
#      Mon Jul 13 21:21:48 2015 -0400
# Node ID 7ddee729f29ee7cb7d82d120fde42129068f3f7d
# Parent  22a2f6a7f6dcb309e8a2cb48787a390cf230f3a4
limit %.class : %.java pattern to specific files

* scripts/java/module.mk (%.class : %.java): Limit pattern to
$(JAVA_CLASSES).

diff --git a/scripts/java/module.mk b/scripts/java/module.mk
--- a/scripts/java/module.mk
+++ b/scripts/java/module.mk
@@ -48,17 +48,17 @@ JAVA_IMAGES = \
 scripts_java_JAVA_SRC = $(addprefix scripts/java/, $(JAVA_SRC))
 
 scripts_java_JAVA_CLASSES = $(addprefix scripts/java/, $(JAVA_CLASSES))
 
 scripts_java_JAVA_IMAGES = $(addprefix scripts/java/, $(JAVA_IMAGES))
 
 srcdir_scripts_java_JAVA_IMAGES = $(addprefix $(srcdir)/scripts/java/, $(JAVA_IMAGES))
 
-%.class : %.java
+$(JAVA_CLASSES) : %.class : %.java
 	$(AM_V_GEN)$(MKDIR_P) scripts/java/$(org_octave_dir) && \
 	( cd $(srcdir)/scripts/java; "$(JAVAC)" -source 1.3 -target 1.3 -d $(abs_top_builddir)/scripts/java $(org_octave_dir)/$(<F) )
 
 scripts/java/images.stamp: $(srcdir_scripts_java_JAVA_IMAGES)
 	$(AM_V_GEN)if [ "x$(srcdir)" != "x." ]; then \
 	  $(MKDIR_P) scripts/java/$(org_octave_dir)/images; \
 	  cp $(srcdir_scripts_java_JAVA_IMAGES) scripts/java/$(org_octave_dir)/images; \
 	fi && \
